spswarehouse/general/selenium.py

```diff
@@ -159,6 +167,14 @@ def type_in_element_by_name(driver, element_name: str, input_to_type: str, wait_
     elem.clear()
     elem.send_keys(input_to_type)
 
+def type_in_element_by_xpath(driver, element_xpath: str, input_to_type: str, wait_time_in_seconds=30):
+    """
+    Waits for an element by XPATH, clears it, and types in the input.
+    """
+    elem =  _wait_for_element_to_be_present_and_return_it(driver, By.XPATH, element_xpath, wait_time_in_seconds)
+    elem.clear()
+    elem.send_keys(input_to_type)
```

Comment:
wondering why this is specifically by XPATH and not more generally by whichever method the helper function you are calling allows?

Reply:
You're asking why `type_in_element_by_xpath()` is specific to XPATH instead of being more general? The goal with these functions is to make it as easy as possible for the caller to just navigate with Selenium while importing as little additional stuff as possible. So if this function were more general like `type_in_element()` and the user had to pass in the `By.XPATH`, then they'd have to import `from selenium.webdriver.common.by import By` as well, beyond just this specific function.

There are potentially some further refactoring optimizations, like making all the `type_in_element_by...` function reuse even more code. We could consider those down the line if we add even more to this file.